From d6063c3e93e9f071a16aa2ffa71ca860ea0b9f92 Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Wed, 6 May 2020 09:23:15 +0200
Subject: [PATCH] net: mdio_device: Reset GPIO is allowed to sleep

Use gpiod_set_value_cansleep in mdio_device_reset since this is called
in a non-atomic context. The solves a kernel WARN message when the
reset GPIO is attached to an I2C IO expander for example.

Signed-off-by: Mike Looijmans <mike.looijmans@topic.nl>
---
 drivers/net/phy/mdio_device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/mdio_device.c b/drivers/net/phy/mdio_device.c
index c924700cf37b..e7f8d14f1d93 100644
--- a/drivers/net/phy/mdio_device.c
+++ b/drivers/net/phy/mdio_device.c
@@ -124,7 +124,7 @@ void mdio_device_reset(struct mdio_device *mdiodev, int value)
 	if (!mdiodev->reset)
 		return;
 
-	gpiod_set_value(mdiodev->reset, value);
+	gpiod_set_value_cansleep(mdiodev->reset, value);
 
 	d = value ? mdiodev->reset_assert_delay : mdiodev->reset_deassert_delay;
 	if (d)
-- 
2.17.1

